pulp_smash/tests/platform/api_v2/test_repository.py

```diff
@@ -0,0 +1,235 @@
+    def setUpClass(cls):
+        """Create several repositories.
+
+        Create one repository with the minimum required attributes, and a
+        second with all available attributes except importers and distributors.
+
+        """
+        cls.cfg = get_config()
+        cls.url = cls.cfg.base_url + REPOSITORY_PATH
+        cls.bodies = (
+            {'id': rand_str()},
+            {
+                'id': rand_str(),
+                'display_name': rand_str(),
+                'description': rand_str(),
+                'notes': {rand_str(): rand_str()},
+            },
+
+        )
+        cls.responses = tuple((
+            requests.post(
+                cls.url,
+                json=body,
+                **cls.cfg.get_requests_kwargs()
+            )
+            for body in cls.bodies
+        ))
+
+    def test_status_code(self):
+        """Assert that each response has a HTTP 201 status code."""
+        for i, response in enumerate(self.responses):
+            with self.subTest(self.bodies[i]):
+                self.assertEqual(response.status_code, 201)
+
+    def test_location_header(self):
+        """Assert that the Location header is correctly set in the response."""
+        for i, response in enumerate(self.responses):
+            with self.subTest(self.bodies[i]):
+                self.assertEqual(
+                    self.url + self.bodies[i]['id'] + '/',
+                    response.headers['Location']
+                )
+
+    def test_attributes(self):
+        """Assert that each repository has the requested attributes."""
+        for i, body in enumerate(self.bodies):
+            with self.subTest(body):
+                attributes = self.responses[i].json()
+                self.assertLessEqual(set(body.keys()), set(attributes.keys()))
+                attributes = {key: attributes[key] for key in body.keys()}
+                self.assertEqual(body, attributes)
+
+    @classmethod
+    def tearDownClass(cls):
+        """Delete the created repositories."""
+        for response in cls.responses:
+            requests.delete(
+                cls.cfg.base_url + response.json()['_href'],
+                **cls.cfg.get_requests_kwargs()
+            ).raise_for_status()
+
+
+class CreateFailureTestCase(TestCase):
+    """Establish that repositories are not created in documented scenarios."""
+
+    @classmethod
+    def setUpClass(cls):
+        """Create several repositories.
+
+        Each repository is created to test a different failure scenario. The
+        first repository is created in order to test duplicate ids.
+
+        """
+        cls.cfg = get_config()
+        cls.url = cls.cfg.base_url + REPOSITORY_PATH
+        identical_id = rand_str()
+        cls.bodies = (
+            (201, {'id': identical_id}),
+            (400, {'id': None}),
+            (400, ['Incorrect data type']),
+            (400, {'missing_required_keys': 'id'}),
+            (409, {'id': identical_id}),
```

Comment:
As per [Exception Handling](https://pulp.readthedocs.org/en/latest/dev-guide/conventions/exceptions.html), these four API calls should return a call report with [`ERROR_KEYS`](http://pulp-smash.readthedocs.org/en/latest/api/pulp_smash.constants.html#pulp_smash.constants.ERROR_KEYS). Can we add in a test method for that?